src/AspNet.Security.OpenId/OpenIdAuthenticationConfiguration.cs

```diff
@@ -221,7 +221,7 @@ where service.Descendants(XName.Get("Type", "xri://$xrd*($v*2.0)"))
                 }
             }
 
-            private Uri ProcessGenericDocument([NotNull] HttpResponseMessage response)
+            private Uri? ProcessGenericDocument(HttpResponseMessage? response)
```

Comment:
Either use `!` at the call sites, or refactor the method to check for `HttpResponseMessage` being `null`. Otherwise it's weird to have an assert for it not being null as the first line of the method when the type is nullable.